src/sparta/checkers/SimpleFlowAnnotatedTypeFactory.java

```diff
@@ -328,6 +328,22 @@ public AnnotationMirror getTopAnnotation(AnnotationMirror start) {
         }
 
         @Override
+        public AnnotationMirror getBottomAnnotation(AnnotationMirror start) {
+            if (start.toString().contains("Sink")) {
```

Comment:
Do you understand why these two methods have a different way of deciding which type system they are? It's the same for getTop, but doesn't really make sense to me. Can you see whether you can treat both more uniformly?